apps/vessel-history/features/activity-by-type/activity-by-type.tsx

```diff
@@ -69,12 +73,17 @@ export function ActivityByType({ onMoveToMap = () => {} }: ActivityByTypeProps)
     [highlightEvent, highlightVoyage, onMoveToMap, setMapCoordinates, viewport.zoom]
   )
 
-  const endDate = DateTime.now()
-  const startDate = endDate.minus(RISK_SUMMARY_SETTINGS.timeRange)
-  const filters: Partial<Filters> = {
-    start: startDate.toUTC().toISO(),
-    end: endDate.toUTC().toISO(),
-  }
+  const filters: Partial<Filters> = useMemo(() => {
+    const endDate =
+      (!online && offlineVessel?.savedOn && DateTime.fromISO(offlineVessel.savedOn)) ||
+      DateTime.now()
+    const startDate = endDate.minus(RISK_SUMMARY_SETTINGS.timeRange)
+
+    return {
+      start: startDate.toUTC().toISO(),
+      end: endDate.toUTC().toISO(),
+    }
+  }, [offlineVessel.savedOn, online])
```

Comment:
this throw an error
`activity-by-type.tsx?a399:86 Uncaught TypeError: Cannot read properties of undefined (reading 'savedOn')
    at ActivityByType (activity-by-type.tsx?a399:86:21)
    at renderWithHooks (react-dom.development.js?3c4a:16305:1)
    at mountIndeterminateComponent (react-dom.development.js?3c4a:20074:1)
    at beginWork (react-dom.development.js?3c4a:21587:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js?3c4a:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?3c4a:4213:1)
    at invokeGuardedCallback (react-dom.development.js?3c4a:4277:1)
    at beginWork$1 (react-dom.development.js?3c4a:27451:1)
    at performUnitOfWork (react-dom.development.js?3c4a:26557:1)
    at workLoopSync (react-dom.development.js?3c4a:26466:1)
    at renderRootSync (react-dom.development.js?3c4a:26434:1)
    at performSyncWorkOnRoot (react-dom.development.js?3c4a:26085:1)
    at flushSyncCallbacks (react-dom.development.js?3c4a:12042:1)
    at eval (react-dom.development.js?3c4a:25651:1)`

@rdgfuentes